lib/AST/Decl.cpp

```diff
@@ -1390,8 +1390,35 @@ unsigned PatternBindingDecl::getPatternEntryIndexForVarDecl(const VarDecl *VD) c
   return ~0U;
 }
 
+Expr *PatternBindingEntry::getOrigInitAsWritten() const {
+  // Before the addition of property wrappers, the following would work:
+  // return InitContextAndIsText.getInt() ? nullptr : InitExpr.Node;
+  // Consider: @Blah(17) var foo: Foo = 18
+  // The code above will return:
+  // Clamped(wrappedValue: 17, min: 0, max: 255)
+  // but what we want is the "18".
+
+  Expr *const init = InitContextAndIsText.getInt() ? nullptr : InitExpr.Node;
+  if (!init)
+    return nullptr;
+  // Assume wrappers only apply to single-variable patterns.
+  VarDecl *const var = getPattern()->getSingleVar();
+  if (!var)
+    return init;
+  auto *e = findOriginalPropertyWrapperInitialValue(var, init);
+  if (e)
+    return e;
+  // Consider: @Wrapper(stored: 17) static var x: Int
+  // It has an "init" but not really a wrapper initial value. Return nullptr in
+  // that case.
+  if (var->getASTContext().SourceMgr.isBeforeInBuffer(init->getStartLoc(),
```

Comment:
This surprised me a bit. When does it happen?

Reply:
I also think this is moot, IIRC.